src/Scalar.jl

```diff
@@ -6,9 +6,15 @@ Construct a statically-sized 0-dimensional array that contains a single element,
 """
 immutable Scalar{T} <: StaticArray{T,0}
     data::T
+
+    Scalar{T}(x::AbstractArray) where {T} = new{T}(convert(T,x))
+    Scalar{T}(x::Tuple{T2}) where {T, T2} = new{T}(convert(T,x[1]))
+    Scalar{T}(x) where {T} = new{T}(convert(T, x))
 end
 
-@inline (::Type{Scalar{T}}){T}(x::Tuple{T}) = Scalar{T}(x[1])
+@inline Scalar(x::Tuple{T}) where {T} = Scalar{T}(x[1])
```

Comment:
The abstract constructors will go `Scalar(x) -> Scalar((x,))`. This then goes to `Scalar{T}(x)`, which is an inner constructor (and is more specific than other constructors).